drivers/display/ssd_1306/ssd_1306.c

```diff
@@ -0,0 +1,178 @@
+ * @brief Initializes the communication peripheral.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		               parameters.
+ *
+ * @return status - Initialization status.
+ *                  Example: -1 - Initialization failed;
+ *                            0 - Initialization succeeded.
+*******************************************************************************/
+int32_t ssd_1306_init(struct ssd_1306_dev **device,
+		 const struct ssd_1306_init_param *param)
+{
+	struct ssd_1306_dev *dev;
+	uint32_t ret;
+
+	dev = (struct ssd_1306_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return -1;
+
+	ret = gpio_get(&dev->reset_pin, param->reset_pin);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_get(&dev->vbat_pin, param->vbat_pin);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_get(&dev->vdd_pin, param->vdd_pin);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_get(&dev->dc_pin, param->dc_pin);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = spi_init(&dev->spi_desc, param->spi_ip);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(dev->dc_pin, DC_CMD);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(dev->vbat_pin, VBAT_ON);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(dev->vdd_pin, VDD_ON);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(dev->reset_pin, RST_OFF);
+	if (ret != SUCCESS)
+		return FAILURE;
+
+	*device = dev;
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Free the resources allocated by ssd_1306_init().
+ *
+ * @param dev - The device structure.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ssd_1306_remove(struct ssd_1306_dev *device)
+{
+	int32_t ret;
+
+	ret = spi_remove(device->spi_desc);
+
+	ret |= gpio_remove(device->dc_pin);
+	ret |= gpio_remove(device->reset_pin);
+	ret |= gpio_remove(device->vbat_pin);
+	ret |= gpio_remove(device->vdd_pin);
+
+	free(device);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Initializes ssd_1306 for display screening.
+ *
+ * @param dev - The device structure.
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t ssd_1306_display_init(struct ssd_1306_dev *device)
+{
+	int32_t ret = SUCCESS;
+	uint8_t command[3];
+	// initial pin state
+	ret |= gpio_set_value(device->reset_pin, RST_OFF);
+	ret |= gpio_set_value(device->dc_pin, DC_CMD);
+	ret |= gpio_set_value(device->vdd_pin, VDD_ON);
+	ret |= gpio_set_value(device->vbat_pin, VBAT_ON);
+	usleep(2000U);
+
+	command[0] = 0xAE;
```

Comment:
Maybe we would benefit from a list of commands as defines or enum in the header file.